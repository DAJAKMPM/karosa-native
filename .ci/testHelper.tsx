import React, { JSXElementConstructor } from "react";
import { View } from "react-native";
import {
  render,
  RenderOptions,
  fireEvent,
  act,
} from "@testing-library/react-native";
import { Formik, FormikValues } from "formik";

export { fireEvent, act };

// for getting findByType e.g. wrapper.findByType(Icon) see implementation in Avatar Component
export const renderWithWrapper = (
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  children: React.ReactElement<any, string | JSXElementConstructor<any>>,
  wrapperTestID?: string,
  _themeProp: unknown = {},
  renderOptions?: RenderOptions
) => {
  const options: RenderOptions = {
    ...(!wrapperTestID && {
      wrapper: (props) => <View {...props} testID="wrapper" />,
    }),
    ...renderOptions,
  };

  const renderApi = render(children, options);
  const wrapper = renderApi.queryByTestId(wrapperTestID || "wrapper");
  return { wrapper, ...renderApi };
};

export const renderWithFormikWrapper = <T extends FormikValues>(
  initialValues: T,
  children: React.ReactElement<any, string | JSXElementConstructor<any>>
) => {
  return renderWithWrapper(
    <Formik
      initialValues={initialValues}
      onSubmit={() => console.log("submit")}
      component={() => children}
    />
  );
};
