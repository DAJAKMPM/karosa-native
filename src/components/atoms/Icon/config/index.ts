/**
 *
 * @format
 *
 */

export const ZERO = 0;
