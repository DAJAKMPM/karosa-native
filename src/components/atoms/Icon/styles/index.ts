/**
 *
 * @format
 *
 */

import {StyleSheet} from 'react-native';

const IconStyles = StyleSheet.create({
});

export default IconStyles;
