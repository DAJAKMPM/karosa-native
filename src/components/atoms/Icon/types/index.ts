/**
 *
 * @format
 *
 */

import { StyleProp, ViewStyle } from "react-native";

export type PropsType = {
  group: string;
  name: string;
  extraStyle: StyleProp<ViewStyle>;
  height: number;
  width: number;
};
