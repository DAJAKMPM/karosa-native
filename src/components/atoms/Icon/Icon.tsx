/**
 *
 * Icon
 * @format
 *
 */

import React, { FC, Fragment } from "react";
import Icons from "@app/assets/icons";

import type { PropsType } from "./types";
import { ZERO } from "./config";

const Icon: FC<PropsType> = (props) => {
  const { group, name, extraStyle, height, width } = props;

  const icon = Icons[group](
    name,
    extraStyle,
    width ? width : ZERO,
    height ? height : ZERO
  );

  return <Fragment>{icon}</Fragment>;
};

export default Icon;
