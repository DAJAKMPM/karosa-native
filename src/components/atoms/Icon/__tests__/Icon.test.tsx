import React from "react";
import { renderWithWrapper } from "@app/ci/testHelper";
import Icon from "../Icon";

describe("Icon Component", () => {
  it("should match snapshot", () => {
    const { toJSON } = renderWithWrapper(
      <Icon group="products" name="chatNow" height={20} width={20} />
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
