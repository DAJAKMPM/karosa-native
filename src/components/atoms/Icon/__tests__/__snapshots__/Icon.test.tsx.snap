// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Component should match snapshot 1`] = `
<View
  testID="wrapper"
>
  <View>
    <SvgMock
      height={20}
      width={20}
    />
  </View>
</View>
`;
